File: desafio-04/JacksonOsvaldo/python/desafio04.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-


"""
Resposta de Jackson Osvaldo da Silva Braga
GitHub: https://github.com/JacksonOsvaldo
E-mail: jacksonosvaldo@live.com
"""


resultado = []
pecas = ['1', '2', '3', '4', '5', '6'] #Peão[1], Bispo[2], Cavalo[3], Torre[4], Rainha[5], Rei[6]
vazio = Peao = Bispo = Cavalo = Torre = Rainha = Rei = 0


def valores (numeros):
    num = numeros
    a = [num.split(' ')]
    return a


def tabuleiro(listValores):
    quantasPecas = []
    contar = 0
    for linha in listValores:
        contar = contar + 1
        for segLin in pecas:
            quantasPecas.append(linha.count(segLin))
    return quantasPecas


for i in range(8):
    pecasDoJogo = input()
    pecasJogo = valores(pecasDoJogo)
    resultado = tabuleiro(pecasJogo)
    Peao = resultado[1] + Peao
    Bispo = resultado[2] + Bispo
    Cavalo = resultado[3] + Cavalo
    Torre = resultado[4] + Torre
    Rainha = resultado[5] + Rainha
    Rei = resultado[6] + Rei

print("""\nPeão: {} peça(s)\nBispo: {} peça(s)\nCavalo: {} peça(s)\nTorre: {} peça(s)\nRainha: {} peça(s)\nRei: {} peça(s)""".format(Peao, Bispo, Cavalo, Torre, Rainha, Rei))

File: desafio-04/itsArthurAlves/cpp/desafio_04.cpp
#include <stdio.h>
#include <stdlib.h>
#include <iostream>
using namespace std;

void entertab(int x[][8]){
  int i, j, w = 9;
  char z;
  for (i=0; i<8; i++){
    w--;
    for(j=0; j<8; j++){
      z = (j==0) ? ('A') :
          (j==1) ? ('B') :
          (j==2) ? ('C') :
          (j==3) ? ('D') :
          (j==4) ? ('E') :
          (j==5) ? ('F') :
          (j==6) ? ('G') :
          (j==7) ? ('H') : ('I');
      cout<<"Vazio: 0\nPeão: 1\nBispo: 2\nCavalo: 3\nTorre: 4\nRainha: 5\nRei: 6\n";
      cout<<"["<<w<<"]"<<"["<<z<<"]"<<" ? : ";
      cin>>x[i][j];
      system("clear");
    };
  }
};

void outtab(int x[][8]){
 int i, j, z, peao = 0, bispo = 0, cavalo = 0, torre = 0, rainha = 0, rei = 0, vazio = 0;
 for(i=0; i<8; i++){
  z = (i==0) ? (8) :
      (i==1) ? (7) :
      (i==2) ? (6) :
      (i==3) ? (5) :
      (i==4) ? (4) :
      (i==5) ? (3) :
      (i==6) ? (2) :
      (i==7) ? (1) : (9);
  cout<<z<<" > ";
  for(j=0;j<8; j++){
     cout<<x[i][j]<<" ";
     vazio = (x[i][j] == 0) ? (vazio+1) : (vazio);
     peao = (x[i][j] == 1) ? (peao+1) : (peao);
     bispo = (x[i][j] == 2) ? (bispo+1) : (bispo);
     cavalo = (x[i][j] == 3) ? (cavalo+1) : (cavalo);
     torre = (x[i][j] == 4) ? (torre+1) : (torre);
     rainha = (x[i][j] == 5) ? (rainha+1) : (rainha);
     rei = (x[i][j] == 6) ? (rei+1) : (rei);
  };
  cout<<"\n";
 };
 cout<<"    ^ ^ ^ ^ ^ ^ ^ ^ \n";
 cout<<"    A B C D E F G H \n";
 cout<<"\nVazio: "<<vazio<<" peça(s)"<<"\nPeão: "<<peao<<" peça(s)\n"<<"Bispo: "<<bispo<<" peça(s)\n"<<"Cavalo: "<<cavalo<<" peça(s)\n";
 cout<<"Torre: "<<torre<<" peça(s)\n"<<"Rainha: "<<rainha<<" peça(s)\n"<<"Rei: "<<rei<<" peça(s)\n";
};

int main(){
  int i, j, matrix[8][8];
  entertab(matrix);
  outtab(matrix);
return(0);
 
}